### ISSUE TITLE
Consider indexing proposals based on on-chain proposal id rather than CGP document number

### ISSUE THREAD

**Issue by m-chrzan** [celo-org/celo-mondo#108]
## Problem

It appears that currently the list of proposals displayed in the Governance tab is based on the CGP documents in the [governance repo](https://github.com/celo-org/governance). This leads to a past on-chain vote getting lost when a proposal is resubmitted with the same referenced document. 

As an example, see CGP-152. It was first put up for voting on November 2 with proposal ID [192](https://celo.stake.id/#/proposal/192), got rejected, then resubmitted on Novermber 29 with ID [195](https://celo.stake.id/#/proposal/195).

The links in the above paragraph are to celo.stake.id, which indexes proposals based on ID, creating a separate page for each on-chain proposal, keeping a history of all Governance activity. Mondo currently does not handle this, and the rejected proposal with ID 192 is nowhere to be found.

## Solution

Index governance proposals based on on-chain ID rather than CGP document number. I understand this could be a bigger reworking, since it appears that on the backend, Mondo currently relies on CGP document numbering for listing out proposals, but I think it's an important change to display a full, transparent history of the Governance process on the site.

**Comment by aaronmgdr** [comment]
One alternative. still index by gcp by display results of both votes on the same page

**Comment by m-chrzan** [comment]
> One alternative. still index by cgp by display results of both votes on the same page

Yep, also a possibility. Though could be confusing in the list view - do we list the CGP twice, or only the most recent one, making the older vote harder to find?

Also, if going with this approach, probably a good idea to generalize from "both" to "all" - so things don't have to be reworked again if we ever get a CGP that goes through the process 3 or more times.

**Comment by aaronmgdr** [comment]
i think fixing so we definitely show the correct one will help

**Comment by aaronmgdr** [comment]
im going to close as wont due as not all proposals will have an on chain id (if they are in draft for instance) and we have instead moved towards showing subling proposals on the same page. we may introduce a proper indexer (just git files atm) which will help with the issues of onchain data not being persistent.
